src/data_types/w3c/credential.rs

```diff
@@ -0,0 +1,367 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::string::ToString;
+use zeroize::Zeroize;
+
+use crate::data_types::w3c::constants::{ANONCREDS_CONTEXTS, ANONCREDS_CREDENTIAL_TYPES};
+use crate::data_types::w3c::credential_proof::{CredentialProof, CredentialSignatureProof};
+use crate::data_types::w3c::presentation_proof::{CredentialPresentationProof, PredicateAttribute};
+use crate::data_types::{
+    cred_def::CredentialDefinitionId,
+    credential::CredentialValuesEncoding,
+    issuer_id::IssuerId,
+    rev_reg_def::RevocationRegistryDefinitionId,
+    schema::SchemaId,
+    w3c::{
+        constants::{
+            W3C_ANONCREDS_CONTEXT, W3C_ANONCREDS_CREDENTIAL_TYPE, W3C_CONTEXT, W3C_CREDENTIAL_TYPE,
+        },
+        one_or_many::OneOrMany,
+        uri::URI,
+    },
+};
+use crate::error::ValidationError;
+use crate::types::{CredentialValues, MakeCredentialValues};
+use crate::utils::validation::Validatable;
+use crate::Result;
+
+/// AnonCreds W3C Credential definition
+/// Note, that this definition is tied to AnonCreds W3C form
+/// Some fields are defined as required despite to general W3C specification
+/// For example `credential_schema` is required for AnonCreds W3C Credentials and has custom format
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct W3CCredential {
+    #[serde(rename = "@context")]
+    pub context: Contexts,
+    #[serde(rename = "type")]
```

Comment:
Not sure that it is really need to be supported.
It can be done through the overriding default deserialization and parsing to an intermediate object first containing both fields. But I do not really see a big sense in doing it.

Reply:
Otherwise we cannot receive all W3cCredentials right? We can do it with `alias` IIRC. https://serde.rs/field-attrs.html#alias